docs/nested_par_for.md

```diff
@@ -0,0 +1,36 @@
+
+# Nested Parallelism
+
+The loop wrappers documented here abstracts a hierarchical parallelism model,
+which allows more fine grain control on core level vs. vector level parallelism
+and allows explicitly defined caches for tightly nested loops. These wrappers
+provide a simplified interface of the [hierarchical parallelism in
+Kokkos](https://github.com/kokkos/kokkos/wiki/HierarchicalParallelism)
```

Comment:
Can you also add a link to the unit test where you use the par fors, so a reader can see an example call?